### PR TITLE
843_CR_Status | Add /status API

### PR DESCRIPTION
**Pull request by bipul-jha**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- consent/src/main/java/in/projecteka/consentmanager/clients/ConsentManagerClient.java  3
- consent/src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  2
- consent/src/test/java/in/projecteka/consentmanager/consent/ConsentArtefactUserJourneyTest.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=6045246&bid=19806388)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- consent/src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  2
- consent/src/test/java/in/projecteka/consentmanager/consent/ConsentArtefactUserJourneyTest.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=6045246&bid=19806388)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- consent/src/test/java/in/projecteka/consentmanager/consent/ConsentArtefactUserJourneyTest.java  5
- consent/src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=6045246&bid=19806388)

File: consent/src/main/java/in/projecteka/consentmanager/clients/ConsentManagerClient.java [hunk 1]
```diff
@@ -94,4 +96,26 @@ public ConsentManagerClient(WebClient.Builder webClient,
...
+                .flatMap(token ->
+                        webClient
+                                .post()
+                                .uri(gatewayServiceProperties.getBaseUrl() + CONSENT_REQUEST_STATUS_URL_PATH)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .header(AUTHORIZATION, token)
+                                .header(HDR_HIU_ID, hiuId)
+                                .bodyValue(consentStatusResponse)
+                                .retrieve()
+                                .onStatus(httpStatus -> httpStatus.value() == 400,
```
[the remark this answers is not in the document]
Comment (MounicaSagi) [reply consent/src/main/java/in/projecteka/consentmanager/clients/ConsentManagerClient.java]: Done.

**Review by mddubey — APPROVED** [review]

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- consent/src/test/java/in/projecteka/consentmanager/consent/ConsentArtefactUserJourneyTest.java  5
- consent/src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=6045246&bid=19806388)

**Pull request merged**